### PR TITLE
Handler streaming prep

### PR DESCRIPTION
**Pull request by jcorbin**
Final handler code rotation before streaming implementation.

reviewers: @Raynos @kriskowal

### CODE REVIEW

**Comment by jcorbin** [comment]
The end of the rainbow comes after this PR:
https://github.com/uber/tchannel/compare/handler_streaming_prep...streaming

File: node/reqres.js [hunk 1]
```diff
@@ -285,21 +287,22 @@ TChannelOutgoingResponse.prototype.sendCallResponseContFrame = function sendCall
...
 };
 
 TChannelOutgoingResponse.prototype.sendErrorFrame = function sendErrorFrame(codeString, message) {
     var self = this;
-    if (self.state === States.Done) {
+    if (self.state === States.Done || self.state === States.Error) {
         throw new Error('response already done'); // TODO: typed error
     } else {
         self.sendFrame.error(codeString, message);
-        self.state = States.Done;
```
Comment (Raynos) [inline node/reqres.js]: Does the protocol state that a call response which is not finished can be terminated by an error response with the same id ?

Should we check that we are not in the Streaming state here ?

Comment (mmihic) [inline node/reqres.js]: Yes you need to be able to fast terminate a response with an Error frame at any time.

**Comment by Raynos** [comment]
Seems legit.

File: node/reqres.js (hunk 1 above)
Comment (Raynos) [inline node/reqres.js]: Opened an issue to have protocol.md updated ( https://github.com/uber/tchannel/issues/234 ).

**Pull request merged**